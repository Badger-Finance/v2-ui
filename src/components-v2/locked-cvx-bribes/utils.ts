import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
