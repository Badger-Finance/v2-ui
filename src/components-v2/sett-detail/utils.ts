import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
