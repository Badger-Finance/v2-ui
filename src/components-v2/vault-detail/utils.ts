import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
