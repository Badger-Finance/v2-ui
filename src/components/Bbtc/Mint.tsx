import React, { useContext, useState } from 'react';

import { Container, Button, Typography } from '@material-ui/core';

import { useStyles, debounce } from './common';
import BigNumber from 'bignumber.js';
import { Tokens } from './Tokens';

import { TokenModel } from './model';

import { StoreContext } from '../../mobx/store-context';

export const Mint = (): any => {
	const store = useContext(StoreContext);
	const classes = useStyles();

	const tokens = store.bbtcStore.tokens;
	const bBTC = store.bbtcStore.bBTC;

	let inputRef: any;

	const [selectedToken, setSelectedToken] = useState<TokenModel>(tokens[0]);
	const [inputAmount, setInputAmount] = useState<number | string>('');
	const _debouncedSetInputAmount = debounce(1000, (val) => {
		setInputAmount(val);
	});

	const handleInputAmount = (event: any) => {
		const nextValue = event?.target?.value;
		_debouncedSetInputAmount(nextValue);
	};

	const handleTokenSelection = (event: any) => {
		const token = tokens.find((token) => token.symbol === event?.target?.value);
		setSelectedToken(token || tokens[0]);
	};

	const useMaxBalance = () => {
		setInputAmount((inputRef.value = selectedToken.balance.toString(10)));
	};
	const handleMintClick = () => {
		store.bbtcStore.mint(selectedToken, selectedToken.scale(new BigNumber(inputAmount)));
	};

	return (
		<Container className={classes.root} maxWidth="lg">
			<div className={classes.outerWrapper}>
				<Typography variant="caption" className={classes.balance}>
					Available {selectedToken.symbol}: {selectedToken.formattedBalance}
				</Typography>
				<div className={classes.inputWrapper}>
					<Tokens tokens={tokens} default={selectedToken.symbol} onTokenSelect={handleTokenSelection} />
					<input
						className={classes.unstylishInput + ' unstylish-input'}
						onChange={handleInputAmount}
						type="number"
						ref={(ref) => (inputRef = ref)}
						min="0"
						placeholder="0.0"
					/>

					<Button className={classes.btnMax} variant="outlined" onClick={useMaxBalance}>
						max
					</Button>
				</div>
			</div>
			<div className={classes.outerWrapper}>
				<svg
					className={classes.arrowDown}
					width="13"
					height="18"
					viewBox="0 0 13 18"
					fill="none"
					xmlns="http://www.w3.org/2000/svg"
				>
					<path
						fillRule="evenodd"
						clipRule="evenodd"
						d="M6.5 15.2138L11.6295 10L13 11.3931L6.5 18L-6.08938e-08 11.3931L1.37054 10L6.5 15.2138Z"
						fill="white"
					/>
					<line x1="6.5" y1="16" x2="6.5" y2="4.37114e-08" stroke="white" strokeWidth="2" />
				</svg>
			</div>
			<div className={classes.outerWrapper}>
				<Typography variant="caption" className={classes.balance}>
					Available {bBTC.symbol}: {bBTC.formattedBalance}
				</Typography>
				<div className={classes.inputWrapper}>
					<div className={classes.token}>
						<img src={bBTC.icon} className={classes.tokenIcon} alt={bBTC.name} />
						<Typography className={classes.tokenLabel} variant="body1">
							{bBTC.symbol}
						</Typography>
					</div>
					<input
						className={classes.unstylishInput + ' unstylish-input'}
						type="number"
						min="0"
						value={inputAmount}
						placeholder="0.0"
						readOnly
					/>
				</div>
			</div>
			<div className={classes.outerWrapper}>
				<div className={classes.summaryWrapper}>
					<div className={classes.summaryRow}>
						<Typography variant="subtitle1">Current Conversion Rate: </Typography>
						<Typography variant="subtitle1">
							1{selectedToken.symbol}: 1{bBTC.symbol}
						</Typography>
					</div>
					<div className={classes.summaryRow}>
						<Typography variant="subtitle1">Fees: </Typography>
						<Typography variant="subtitle1">0.002 {bBTC.symbol}</Typography>
					</div>
				</div>
			</div>
			<div className={classes.outerWrapper}>
				<Button size="large" variant="contained" color="primary" onClick={handleMintClick}>
					MINT
				</Button>
			</div>
		</Container>
	);
};
