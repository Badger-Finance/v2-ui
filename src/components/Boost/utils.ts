import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
