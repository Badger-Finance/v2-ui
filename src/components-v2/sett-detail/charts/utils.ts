import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
