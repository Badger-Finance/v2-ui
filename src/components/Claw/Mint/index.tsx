import React from 'react';
import { Grid, Box } from '@material-ui/core';
import { ethers } from 'ethers';
import { observer } from 'mobx-react-lite';

import { StoreContext } from 'mobx/store-context';
import { TokenSelectorLabel } from 'components-v2/common/TokenSelectorLabel';
import { TokenSelect } from 'components-v2/common/TokenSelect';
import { TokenAmountInput } from 'components-v2/common/TokenAmountInput';
import { PercentageGroup } from 'components-v2/common/PercentageGroup';
import { TokenSelectorWithAmountContainer } from 'components-v2/common/TokenSelectorWithAmountContainer';
import { ActionButton } from '../ActionButton';
import { ClawDetails } from '../ClawDetails';
import { useError, useMaxClaw, useDetails, useValidateClaw } from './mint.hooks';
import { mintReducer, State } from './mint.reducer';
import { scaleToString, Direction } from 'utils/componentHelpers';

const initialState: State = { collateral: {}, synthetic: {} };

export const Mint = observer(() => {
	const { claw: store, contracts, wallet } = React.useContext(StoreContext);
	const { collaterals, clawsByCollateral, syntheticsDataByEMP } = store;
	const [state, dispatch] = React.useReducer(mintReducer, initialState);
	const { collateral, synthetic } = state;
	const error = useError(collateral, synthetic);
	const maxClaw = useMaxClaw(collateral, synthetic);
	const mintDetails = useDetails(collateral, synthetic);
	const validateClaw = useValidateClaw(synthetic);

	const collateralToken = contracts.tokens[collateral.selectedOption || ''];

	const handleMint = async () => {
		const [empAddress, mintAmount] = [synthetic.selectedOption, synthetic.amount];
		const [collateralAddress, collateralAmount] = [collateral.selectedOption, collateral.amount];
		const decimals: number | undefined = contracts.tokens[collateralAddress || '']?.decimals;

		if (!empAddress || !mintAmount || !decimals || !collateralAmount) return;

		await store.actionStore.mint(
			empAddress,
			ethers.utils.parseUnits(collateralAmount, decimals).toHexString(),
			ethers.utils.parseUnits(mintAmount, decimals).toHexString(),
		);
	};

	return (
		<Grid container>
			<Box clone pb={4}>
				<Grid item xs={12}>
					<TokenSelectorWithAmountContainer
						tokenBalanceInformation={
							<TokenSelectorLabel
								name="Collateral"
								balanceLabel={
									collateralToken && `Available ${collaterals.get(collateralToken.address)}`
								}
								balance={
									collateralToken?.balance &&
									scaleToString(collateralToken.balance, collateralToken.decimals, Direction.Down)
								}
							/>
						}
						tokenList={
							<TokenSelect
								selectedOption={collateral.selectedOption}
								options={collaterals}
								placeholder="Select Token"
								disabled={!wallet.connectedAddress || collaterals.size === 0}
								onChange={(selectedOption: string) => {
									dispatch({ type: 'COLLATERAL_OPTION_CHANGE', payload: selectedOption });
								}}
							/>
						}
						tokenAmount={
							<TokenAmountInput
								value={collateral.amount}
								disabled={!collateral.selectedOption}
								onChange={(amount: string) => {
									if (!collateralToken) return;
									dispatch({
										type: 'COLLATERAL_AMOUNT_CHANGE',
										payload: { amount, collateralToken },
									});
								}}
							/>
						}
						percentagesGroup={
							<PercentageGroup
								disabled={!collateral.selectedOption}
								options={[25, 50, 75, 100]}
								onChange={(percentage: number) => {
									if (!collateralToken) return;
									dispatch({
										type: 'COLLATERAL_PERCENTAGE_CHANGE',
										payload: { percentage, collateralToken },
									});
								}}
							/>
						}
					/>
				</Grid>
			</Box>
			<Grid item xs={12}>
				<TokenSelectorWithAmountContainer
					tokenBalanceInformation={
						<TokenSelectorLabel
							name="Mintable"
							balanceLabel={maxClaw ? 'Maximum CLAW:' : ''}
							balance={
								maxClaw &&
								collateralToken &&
								scaleToString(maxClaw, collateralToken.decimals, Direction.Down)
							}
						/>
					}
					tokenList={
						<TokenSelect
							placeholder="Select CLAW"
							selectedOption={synthetic.selectedOption}
							options={
								collateral.selectedOption ? clawsByCollateral.get(collateral.selectedOption) : undefined
							}
							disabled={!collateral.selectedOption || !collateral.amount}
							onChange={(selectedOption: string) =>
								dispatch({ type: 'SYNTHETIC_OPTION_CHANGE', payload: selectedOption })
							}
						/>
					}
					tokenAmount={
						<TokenAmountInput
							value={synthetic.amount}
							disabled={!collateral.selectedOption || !synthetic.selectedOption}
							onChange={(amount: string) => {
								const syntheticData = syntheticsDataByEMP.get(synthetic.selectedOption || '');
								if (!collateralToken || !syntheticData) return;
								dispatch({
									type: 'SYNTHETIC_AMOUNT_CHANGE',
									payload: { amount, collateralToken, syntheticData },
								});
							}}
						/>
					}
					percentagesGroup={
						<PercentageGroup
							disabled={!collateral.selectedOption || !synthetic.selectedOption}
							options={[25, 50, 75, 100]}
							onChange={(percentage: number) => {
								const syntheticData = syntheticsDataByEMP.get(synthetic.selectedOption || '');
								if (!syntheticData || !maxClaw || !collateralToken) return;
								dispatch({
									type: 'SYNTHETIC_PERCENTAGE_CHANGE',
									payload: {
										percentage,
										maxClaw,
										syntheticData,
										collateralToken,
										validateClaw,
									},
								});
							}}
						/>
					}
				/>
			</Grid>
			<Grid item xs={12}>
				<ClawDetails details={mintDetails} />
			</Grid>
			<Grid item xs={12}>
				<Grid container>
					<ActionButton
						text={error ? error : 'MINT'}
						onClick={handleMint}
						disabled={!!error || !collateral.selectedOption || !synthetic.selectedOption}
					/>
				</Grid>
			</Grid>
		</Grid>
	);
});

export default Mint;
