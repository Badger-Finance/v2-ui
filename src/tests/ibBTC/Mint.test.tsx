/* eslint-disable @typescript-eslint/no-empty-function */
import '@testing-library/jest-dom';

import { ibBTCService, TransactionStatus } from '@badger-dao/sdk';
import { cleanup, fireEvent, within } from '@testing-library/react';
import { parseEther } from 'ethers/lib/utils';
import React from 'react';
import { ToastContainer } from 'react-toastify';

import { Mint } from '../../components/IbBTC/Mint';
import { TokenBalance } from '../../mobx/model/tokens/token-balance';
import store from '../../mobx/stores/RootStore';
import { StoreProvider } from '../../mobx/stores/store-context';
import { customRender, screen } from '../Utils';
import { SAMPLE_IBBTC_TOKEN_BALANCE, SAMPLE_IBBTC_USER_BALANCES } from '../utils/samples';

jest.useFakeTimers();

describe('ibBTC Mint', () => {
  beforeEach(() => {
    store.wallet.address = '0x1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a';
    store.user.balances = SAMPLE_IBBTC_USER_BALANCES;
    store.ibBTCStore.mintFeePercent = 0;
    store.ibBTCStore.redeemFeePercent = 0;

    store.ibBTCStore.mintRates = {
      '0x6dEf55d2e18486B9dDfaA075bc4e4EE0B28c1545': '1.024385',
      '0xEB4C2781e4ebA804CE9a9803C67d0893436bB27D': '0.992518',
      '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599': '0.992540',
      '0xd04c48A53c111300aD41190D63681ed3dAd998eC': '1.016724',
      '0xb9D076fDe463dbc9f915E5392F807315Bf940334': '1.030187',
      '0x4b92d19c11435614CD49Af1b589001b7c08cD4D5': '1.008024',
      '0x8c76970747afd5398e958bDfadA4cf0B9FcA16c4': '1.001849',
      '0x5Dce29e92b1b939F8E8C60DcF15BDE82A85be4a9': '1.003153',
      '0xf349c0faA80fC1870306Ac093f75934078e28991': '0.999124',
      '0x55912D0Cf83B75c492E761932ABc4DB4a5CB1b17': '0.999502',
    };

    store.ibBTCStore.redeemRates = {
      '0xd04c48A53c111300aD41190D63681ed3dAd998eC': '1.024385',
    };

    /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
    jest.spyOn(ibBTCService.prototype, 'estimateMint').mockImplementation(async (_balance) => ({
      bbtc: TokenBalance.fromBalance(SAMPLE_IBBTC_TOKEN_BALANCE, 11.988).tokenBalance,
      fee: TokenBalance.fromBalance(SAMPLE_IBBTC_TOKEN_BALANCE, 0.012).tokenBalance,
    }));
  });

  afterEach(cleanup);

  it('displays token input balance and ibBTC balance', () => {
    customRender(
      <StoreProvider value={store}>
        <Mint />
      </StoreProvider>,
    );
    expect(screen.getByText('Balance: 5.00000000')).toBeInTheDocument();
    expect(screen.getByText('Balance: 10.000000')).toBeInTheDocument();
  });

  it('can apply max balance', async () => {
    const { container } = customRender(
      <StoreProvider value={store}>
        <Mint />
      </StoreProvider>,
    );

    fireEvent.click(await screen.findByRole('button', { name: /max/i }));
    await screen.findByText('11.988000');
    expect(container).toMatchSnapshot();
  });

  it('displays output ibBTC when mint amount is inputted', async () => {
    const { container } = customRender(
      <StoreProvider value={store}>
        <Mint />
      </StoreProvider>,
    );

    fireEvent.change(await screen.findByRole('textbox'), { target: { value: '12' } });
    await screen.findByRole('heading', { level: 3, name: '11.988000' });

    expect(container).toMatchSnapshot();
  });

  it('can change token', async () => {
    jest.useRealTimers();
    store.api.loadProtocolSummary = jest.fn();
    const { container } = customRender(
      <StoreProvider value={store}>
        <Mint />
      </StoreProvider>,
    );

    fireEvent.mouseDown(screen.getByRole('button', { name: store.ibBTCStore.mintOptions[0].token.symbol }));

    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: store.ibBTCStore.mintOptions[1].token.symbol,
      }),
    );

    await screen.findByText(store.ibBTCStore.mintOptions[1].token.symbol);
    await screen.findByDisplayValue(store.ibBTCStore.mintOptions[1].token.address);

    expect(container).toMatchSnapshot();
  });

  it('handles empty balance', async () => {
    jest.useFakeTimers();

    customRender(
      <StoreProvider value={store}>
        <ToastContainer position="bottom-right" newestOnTop={true} closeOnClick theme="dark" draggable />
        <Mint />
      </StoreProvider>,
    );

    fireEvent.change(screen.getByRole('textbox'), { target: { value: '12' } });

    jest.runAllTimers();

    await screen.findByText('11.988000 ibBTC');

    fireEvent.click(screen.getByRole('button', { name: /mint/i }));

    await screen.findByText('You have insufficient balance of bcrvRenWBTC');
  });

  it('executes calcMint with correct params', async () => {
    const calcMintSpy = jest.spyOn(ibBTCService.prototype, 'estimateMint');

    customRender(
      <StoreProvider value={store}>
        <Mint />
      </StoreProvider>,
    );

    fireEvent.change(screen.getByRole('textbox'), { target: { value: '0.1' } });

    jest.runAllTimers();

    await screen.findByText('11.988000 ibBTC');

    expect(calcMintSpy).toHaveBeenNthCalledWith(1, store.ibBTCStore.mintOptions[0].token.address, parseEther('0.1'));
  });

  it('executes mint with correct params', async () => {
    const mintSpy = jest
      .spyOn(ibBTCService.prototype, 'mint')
      .mockReturnValue(Promise.resolve(TransactionStatus.Success));

    customRender(
      <StoreProvider value={store}>
        <Mint />
      </StoreProvider>,
    );

    fireEvent.change(screen.getByRole('textbox'), { target: { value: '0.1' } });

    jest.runAllTimers();

    await screen.findByText('11.988000 ibBTC');

    fireEvent.click(screen.getByRole('button', { name: /mint/i }));

    await screen.findByDisplayValue('');

    expect.extend({
      toHaveCorrectNonFunctionParams(received, expected) {
        const pass =
          received.token === expected.token &&
          +received.amount === +expected.amount &&
          received.slippage === expected.slippage;

        if (pass) {
          return {
            message: () => `expected ${received} to have correct non-function params`,
            pass: true,
          };
        } else {
          return {
            message: () => `expected ${received} to have correct non-function params`,
            pass: false,
          };
        }
      },
    });

    expect(mintSpy.mock.calls[0][0]).toHaveCorrectNonFunctionParams({
      token: store.ibBTCStore.mintOptions[0].token.address,
      amount: parseEther('0.1'),
      slippage: 1,
    });
  });
});
