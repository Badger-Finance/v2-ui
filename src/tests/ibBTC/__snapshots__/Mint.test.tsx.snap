// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ibBTC Mint displays output ibBTC when mint amount is inputted 1`] = `
<div>
  <div
    class="MuiGrid-root-1 MuiGrid-container-2"
  >
    <div
      class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-104 WithStyles(ForwardRef(Grid))-root-105 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <p
        class="MuiTypography-root-107 WithStyles(ForwardRef(Typography))-root-106 MuiTypography-body1-109 MuiTypography-colorTextSecondary-133"
      >
        Balance: 
        5.000
      </p>
    </div>
    <div
      class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-137 WithStyles(ForwardRef(Grid))-root-138 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47 MuiGrid-grid-sm-5-54"
      >
        <div
          class="MuiInputBase-root-141 NumericInput-root-139 makeStyles-input-140 MuiInputBase-disabled-144 MuiInputBase-fullWidth-151"
        >
          <input
            class="MuiInputBase-input-152 MuiInputBase-disabled-144"
            disabled=""
            pattern="^[0-9]*[.,]?[0-9]*$"
            placeholder="0.000"
            type="tel"
            value="12"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-159 WithStyles(ForwardRef(Grid))-root-160 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-item-3 MuiGrid-grid-xs-12-47 MuiGrid-grid-sm-7-56"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <button
            class="MuiButtonBase-root-190 MuiButton-root-161 MuiButton-outlined-166 MuiButton-outlinedSizeSmall-178 MuiButton-sizeSmall-182"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-162"
            >
              max
            </span>
            <span
              class="MuiTouchRipple-root-219"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <button
            class="MuiButtonBase-root-190 MuiButton-root-161 MuiButton-outlined-166 makeStyles-selectButton-198 MuiButton-outlinedSizeSmall-178 MuiButton-sizeSmall-182"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-162"
            >
              <div
                style="align-items: center; display: flex; flex-wrap: nowrap; overflow: hidden;"
              >
                <img
                  alt="Badger crvRenWSBTC"
                  style="height: 2rem; margin-right: .2rem; display: block;"
                />
                <div
                  class="MuiTypography-root-107 MuiTypography-body1-109"
                >
                  bcrvRenWSBTC
                </div>
              </div>
              <span
                class="MuiButton-endIcon-186 MuiButton-iconSizeSmall-187"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-200"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-219"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-align-items-xs-center-10 MuiGrid-grid-xs-12-47"
  >
    <svg
      class="makeStyles-arrowDown-209"
      fill="none"
      height="18"
      viewBox="0 0 13 18"
      width="13"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M6.5 15.2138L11.6295 10L13 11.3931L6.5 18L-6.08938e-08 11.3931L1.37054 10L6.5 15.2138Z"
        fill="white"
        fill-rule="evenodd"
      />
      <line
        stroke="white"
        stroke-width="2"
        x1="6.5"
        x2="6.5"
        y1="16"
        y2="4.37114e-08"
      />
    </svg>
  </div>
  <div
    class="MuiGrid-root-1 MuiGrid-container-2"
  >
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <p
        class="MuiTypography-root-107 WithStyles(ForwardRef(Typography))-root-210 WithStyles(ForwardRef(Typography))-root-211 MuiTypography-body1-109 MuiTypography-colorTextSecondary-133"
      >
        Balance: 
        10.000
      </p>
    </div>
    <div
      class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-212 WithStyles(ForwardRef(Grid))-root-213 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47 MuiGrid-grid-sm-9-58 MuiGrid-grid-md-12-75 MuiGrid-grid-lg-10-87"
      >
        <h1
          class="MuiTypography-root-107 WithStyles(ForwardRef(Typography))-root-214 WithStyles(ForwardRef(Typography))-root-215 MuiTypography-h1-112"
        >
          11.988
        </h1>
      </div>
      <div
        class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-216 WithStyles(ForwardRef(Grid))-root-217 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-grid-xs-12-47 MuiGrid-grid-sm-3-52 MuiGrid-grid-md-12-75 MuiGrid-grid-lg-2-79"
      >
        <div
          style="align-items: center; display: flex; flex-wrap: nowrap; overflow: hidden;"
        >
          <img
            alt="Interest-Bearing BTC"
            style="height: 2rem; margin-right: .2rem; display: block;"
          />
          <div
            class="MuiTypography-root-107 MuiTypography-body1-109"
          >
            ibBTC
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
  >
    <div
      class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-218"
    >
      <div
        class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-justify-xs-space-between-21 MuiGrid-grid-xs-12-47"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-6-41"
        >
          <h6
            class="MuiTypography-root-107 MuiTypography-subtitle1-118"
          >
            Current Conversion Rate: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-6-41"
        >
          <p
            class="MuiTypography-root-107 WithStyles(ForwardRef(Typography))-root-106 MuiTypography-body1-109"
          >
            1 
            bcrvRenWSBTC
             : 
            1
             
            ibBTC
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-justify-xs-space-between-21 MuiGrid-grid-xs-12-47"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-6-41"
        >
          <h6
            class="MuiTypography-root-107 MuiTypography-subtitle1-118"
          >
            Fees: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-6-41"
        >
          <p
            class="MuiTypography-root-107 WithStyles(ForwardRef(Typography))-root-106 MuiTypography-body1-109"
          >
            0.012
             
            ibBTC
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-justify-xs-space-between-21 MuiGrid-grid-xs-12-47"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-6-41"
        >
          <h6
            class="MuiTypography-root-107 MuiTypography-subtitle1-118"
          >
            Total Mint Amount: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-6-41"
        >
          <p
            class="MuiTypography-root-107 WithStyles(ForwardRef(Typography))-root-106 MuiTypography-body1-109"
          >
            11.988 ibBTC
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
  >
    <button
      class="MuiButtonBase-root-190 MuiButton-root-161 MuiButton-contained-169 MuiButton-containedPrimary-170 MuiButton-containedSizeLarge-181 MuiButton-sizeLarge-183 MuiButton-fullWidth-184"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-162"
      >
        Connect Wallet
      </span>
      <span
        class="MuiTouchRipple-root-219"
      />
    </button>
  </div>
</div>
`;
